environs/config/config.go

```diff
@@ -1273,6 +1273,17 @@ func (cfg *Config) ValidateUnknownAttrs(fields schema.Fields, defaults schema.De
 				logger.Warningf("unknown config field %q", name)
 			}
 			result[name] = value
+			// The only allowed types for unknown attributes are string, int and bool
+			switch value.(type) {
+			case string:
+				continue
+			case int:
+				continue
+			case bool:
+				continue
+			default:
```

Comment:
I would have thought we would at least support float64. I don't think JSON/YAML will automatically create a uint unless you've used schema to Coerce it into one. But I think often just
foo: 1

can actually create a float64